Observer/TrackingSaveObserver.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Bolt\Boltpay\Observer;
+
+use Bolt\Boltpay\Helper\Api as ApiHelper;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Framework\Event\Observer;
+use Bolt\Boltpay\Helper\Config as ConfigHelper;
+use Magento\Framework\DataObjectFactory;
+use Bolt\Boltpay\Helper\Bugsnag;
+use Bolt\Boltpay\Helper\MetricsClient;
+use Bolt\Boltpay\Helper\FeatureSwitch\Decider;
+
+/**
+ * Class TrackingSaveObserver
+ *
+ * @package Bolt\Boltpay\Observer
+ */
+class TrackingSaveObserver implements ObserverInterface
+{
+    /**
+     * @var ConfigHelper
+     */
+    private $configHelper;
+
+    /**
+     * @var DataObjectFactory
+     */
+    private $dataObjectFactory;
+
+    /**
+     * @var ApiHelper
+     */
+    private $apiHelper;
+
+    /**
+     * @var Bugsnag
+     */
+    private $bugsnag;
+
+    /**
+     * @var MetricsClient
+     */
+    private $metricsClient;
+
+    /**
+     * @var Decider
+     */
+    private $decider;
+
+    /**
+     * @param ConfigHelper      $configHelper
+     * @param DataObjectFactory $dataObjectFactory
+     * @param ApiHelper $apiHelper
+     * @param Bugsnag $bugsnag
+     * @param MetricsClient $metricsClient
+     *
+     */
+    public function __construct(
+        ConfigHelper $configHelper,
+        DataObjectFactory $dataObjectFactory,
+        ApiHelper $apiHelper,
+        Bugsnag $bugsnag,
+        MetricsClient $metricsClient,
+        Decider $decider
+    ) {
+        $this->configHelper = $configHelper;
+        $this->dataObjectFactory = $dataObjectFactory;
+        $this->apiHelper = $apiHelper;
+        $this->bugsnag = $bugsnag;
+        $this->metricsClient = $metricsClient;
+        $this->decider = $decider;
+    }
+
+    /**
+     * @param Observer $observer
+     */
+    public function execute(Observer $observer)
+    {
+        if (!$this->decider->isTrackingSaveEventsEnabled()) {
+            return;
+        }
+
+        try {
+            $startTime = $this->metricsClient->getCurrentTime();
+            $tracking = $observer->getEvent()->getTrack();
+            $shipment = $tracking->getShipment();
+            $order = $shipment->getOrder();
+            $payment = $order->getPayment();
```

Comment:
@akulkapoor a better way of checking if it is not a bolt payment (or no payment at all):
```php
!$payment || $payment->getMethod() != \Bolt\Boltpay\Model\Payment::METHOD_CODE
```